pilot/pkg/model/push_context.go

```diff
@@ -613,6 +617,31 @@ func (ps *PushContext) initVirtualServices(env *Environment) error {
 	return nil
 }
 
+// Caches list of Sidecar resources
+func (ps *PushContext) initSidecarScopes(env *Environment) error {
+	sidecarConfigs, err := env.List(Sidecar.Type, NamespaceAll)
+	if err != nil {
+		return err
+	}
+
+	sortConfigByCreationTime(sidecarConfigs)
+
+	for _, sidecarConfig := range sidecarConfigs {
+		// TODO: add entries with workloadSelectors first before adding namespace-wide entries
+		ps.sidecarsByNamespace[sidecarConfig.Namespace] = append(ps.sidecarsByNamespace[sidecarConfig.Namespace], ConvertToSidecarScope(ps, &sidecarConfig))
+	}
+
+	// prebuild default sidecar scopes for other namespaces
+	for _, s := range ps.ServiceByHostname {
+		ns := s.Attributes.Namespace
+		if len(ps.sidecarsByNamespace[ns]) == 0 {
+			ps.sidecarsByNamespace[ns] = []*SidecarScope{DefaultSidecarScopeForNamespace(ps, ns)}
```

Comment:
We may have proxy not co-located with an service instance? Then this `sidecarsByNamespace` can not cover all namespaces of the mesh proxies.

Reply:
Find answer in `GetSidecarScope`